### PR TITLE
Feat/1461 update

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

Update package description and some dependencies to use `workspace`
inheritance.

* Also updates circleci to produce dockerhub image.

## Testing

No functional changes

## Issue(s)

Closes https://github.com/mozilla-services/syncstorage-rs/issues/1461
Issue https://github.com/mozilla-services/syncstorage-rs/issues/1362

### CODE REVIEW

File: syncserver-common/src/metrics.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use std::collections::HashMap;
 use std::net::UdpSocket;
+use std::sync::Arc;
```
Comment (mhammond) [inline syncserver-common/src/metrics.rs:3]: did functionality changes sneak in here accidentally?

**Review by mhammond — COMMENTED** [review]
Eeek, sorry I left this hanging, but I missed it entirely! While the workspace changes looks fine, there's also functionality changes here which I can't really review.

File: syncserver-common/src/metrics.rs (hunk 1 above)
Comment (jrconlin) [reply syncserver-common/src/metrics.rs:3]: A recent change in StatsdClient changed the way that it is supposed to be called. Previously, you would clone the client, but now, it's a singleton that needs to be reference counted.

**Comment by pjenvey** [comment]
Per https://github.com/mozilla-services/syncstorage-rs/issues/1362#issuecomment-1430543424 the "latest" tag change made to the cirleci config was [originally made by SRE](https://github.com/mozilla-services/syncstorage-rs/pull/488), looping them in as reviewers.

File: .circleci/config.yml [hunk 2]
```diff
@@ -363,7 +363,7 @@ jobs:
           command: |
             export UTILS_DOCKERHUB_REPO=mozilla/sync-spanner-py-utils
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
-              DOCKER_TAG="${CIRCLE_SHA1}"
+              DOCKER_TAG="latest"
```
Comment (cvalaas) [inline .circleci/config.yml:366]: could,should we make the tags "latest,${CIRCLE_SHA1}" ?

Comment (cvalaas) [reply .circleci/config.yml:366]: (I don't know if that's a valid syntax...)

Comment (cvalaas) [reply .circleci/config.yml:366]: It's not, no. Okay - well, looks like the SHA tag is only used to trigger dev builds, which I don't think is a thing we're using or even looking at, so I'd say we could switch back to latest.

**Review by cvalaas — APPROVED** [review]

**Review by cvalaas — APPROVED** [review]

**Pull request merged**